kernel/scheduler.c

```diff
@@ -219,6 +222,8 @@ struct __attribute__((packed)) task_block {
 
     /* threads */
     struct task **threads;
+    struct task *joined_thread;
```

Comment:
This should possibly be "joiner_thread" instead, as we are storing which thread is the joiner of the current one.
Also, what happens if the thread dies and the pointer is reused? It might be safer if we store pid/tid instead.